forge/test/mlir/mnist/training/test_training.py

```diff
@@ -21,6 +21,92 @@
 def test_mnist_training():
     torch.manual_seed(0)
 
+    # Model and data type. To use bfloat16 follow the instructions found at test_forge_vs_torch() in this file.
+    dtype = torch.float32
+
+    # Set training hyperparameters
+    num_epochs = 3
+    batch_size = 64
+    learning_rate = 0.001
+
+    # Limit number of batches to run - quicker test
+    limit_num_batches = 1000
+
+    # Load dataset
+    test_loader, train_loader = load_dataset(batch_size, dtype=dtype)
+
+    # Define model and instruct it to compile and run on TT device
+    framework_model = MNISTLinear(
+        bias=False, dtype=dtype
+    )  # bias=False because batch_size=1 with bias=True is not supported
+
+    # Create a torch loss and leave on CPU
+    loss_fn = torch.nn.CrossEntropyLoss()
+
+    # Define optimizer and instruct it to compile and run on TT device
+    framework_optimizer = torch.optim.SGD(framework_model.parameters(), lr=learning_rate)
+    tt_model = forge.compile(
+        framework_model,
+        sample_inputs=[torch.rand(batch_size, 784, dtype=dtype)],
+        optimizer=framework_optimizer,
+        training=True,
+    )
+
+    logger.info("Starting training loop... (logger will be disabled)")
+    logger.disable("")
+    for epoch_idx in range(num_epochs):
+
+        total_loss = 0
+        for batch_idx, (data, target) in enumerate(train_loader):
+            # Reset gradients (every batch)
+            framework_optimizer.zero_grad()
+
+            # Create target tensor and leave on CPU
+            target = nn.functional.one_hot(target, num_classes=10).to(dtype)
+
+            # Forward pass (prediction) on device
+            pred = tt_model(data)[0]
+            golden_pred = framework_model(data)
+            assert golden_pred.dtype == dtype
+            assert compare_with_golden(golden_pred, pred, verify_cfg=VerifyConfig(pcc=0.95))
```

Comment:
After this [PR](https://github.com/tenstorrent/tt-forge-fe/pull/739), there is a unified method for comparing golden output with output of compiled model: [linkt to verify](https://github.com/tenstorrent/tt-forge-fe/commit/17442e9aa4303dd6189964cffa9d2fc9262fb472#diff-e20a777f7a22e33ebefaedc11a005cbb40c7ca37adbe814a5d84baeae45e775bR266). It calls this compare_with_golden but also performs dataformat and some other sanity checks...

In this PR above, you can find examples of its use in tests. It makes things more concise...

Reply:
I need the predictions from the models since I am calculating the loss after this.
Should I change the `verify` method to make it return the predictions?

Reply:
Verify is under development at the moment (by @vkovinicTT), so I wouldn't change it. Let's leave `compare_with_golden` then...
@vkovinicTT did you plan on adding return for the predictions any time soon?

Reply:
Let's not diverge and extend the scope of planned changes for the verify refactor, yet. There are not so many training tests, so it's not a big problem at this moment IMO...

Reply:
I didn't plan on extending it, but I can do that if needed?